schema.ts

```diff
@@ -1135,6 +1134,26 @@ export interface IndexDefinition {
   popularityRankProperty?: PropertyIdentifier<string>;
 }
 
+/**
+ * Defines the parent of an object, which controls various behaviors for this object.
+ * TODO(alexd): Unhide this
+ * @hidden
+ */
+export interface ParentDefinition {
+  /**
+   * The name of the property within {@link ObjectSchemaDefinition.properties} that
+   * identifies the parent of this object.
+   *
+   * ValueType.Array is not supported is not supported for parentIdProperty.
+   *
+   * If this property references a property with {@link ValueType.Object} and
+   * {@link ValueHintType.Reference}, then it refers to an object in another logical table.
+   *
+   * Otherwise, it refers to an object in the same table.
+   */
+  parentIdProperty: PropertyIdentifier<string>;
+}
```

Comment:
Do you not want to play with the bits for `inheritsPermissions` and `inheritsLifecycle` or whatever? Or first draft is to see what happens without those?

Reply:
My initial SWAG was to encode whatever behavior I wanted when I used these and systematize them as we used it in more places. But happy to plug in something if you prefer. :)

Reply:
That's fine, can add it in later.